src/app/files/browser-files.js

```diff
@@ -140,40 +119,57 @@ function Files (storage) {
   //   }
   // }
   //
-  this.listAsTree = function () {
-    function hashmapize (obj, path, val) {
-      var nodes = path.split('/')
-      var i = 0
-
-      for (; i < nodes.length - 1; i++) {
-        var node = nodes[i]
-        if (obj[node] === undefined) {
-          obj[node] = {}
+  this.resolveDirectory = function (path, callback) {
+    var self = this
+    // path = '' + (path || '')
+    setTimeout(function () {
```

Comment:
why `setTimeout` here?

Reply:
to make `browser-files.js` and `shared-folder.js` work the same.
The latter returns in the next tick, because it is really asynchronous.
The former returns syncronous in the same tick without `setTimeout`